code/game/gamemodes/miniantags/guardian/guardian.dm

```diff
@@ -182,15 +182,18 @@
 		input = stripped_input(src, "Please enter a message to tell your summoner.", "Guardian", "")
 	else
 		input = message
-	if(!input) return
+	if(!input)
+		return
 
-	for(var/mob/M in GLOB.mob_list)
-		if(M == summoner)
-			to_chat(M, "<span class='changeling'><i>[src]:</i> [input]</span>")
-			log_say("(GUARDIAN to [key_name(M)]) [input]", src)
-		else if(M in GLOB.dead_mob_list && M.client && M.stat == DEAD && !isnewplayer(M))
-			to_chat(M, "<span class='changeling'><i>Guardian Communication from <b>[src]</b> ([ghost_follow_link(src, ghost=M)]): [input]</i>")
+	// Show the message to the host and to the guardian.
+	to_chat(summoner, "<span class='changeling'><i>[src]:</i> [input]</span>")
 	to_chat(src, "<span class='changeling'><i>[src]:</i> [input]</span>")
+	log_say("(GUARDIAN to [key_name(summoner)]) [input]", src)
```

Comment:
Might be nice to add a log here using the new system as well.
`create_log(SAY_LOG, "GUARDIAN say: [input]", target)` for example.

Reply:
Oh did no know about this. Should I add this as an additional thing or a replacement for this line?

Reply:
Addition. The `log_say` will write it to a log file.
`create_log` will create a log on the mob you're calling it from (src in this case). Which is saved in memory only

Reply:
Ok, and I'm assuming the `target` should be `summoner` since he is the target of the communication here.

Reply:
Indeed so. This is a one way log. It won't create a log for the target.